From 7f12e956c35a863239855d2eaa8c1b9f130e7a5b Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Wed, 16 Oct 2013 13:58:31 +0800
Subject: [PATCH 35/50] ENGR00266234 remove audio adjustment and uncomment
 network dropping packet.

remove audio drop patch and uncomment network dropping packet.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 .../libstagefright/foundation/ANetworkSession.cpp  |  2 +-
 media/libstagefright/wifi-display/MediaSender.cpp  | 42 ----------------------
 media/libstagefright/wifi-display/MediaSender.h    |  1 -
 3 files changed, 1 insertion(+), 44 deletions(-)

diff --git a/media/libstagefright/foundation/ANetworkSession.cpp b/media/libstagefright/foundation/ANetworkSession.cpp
index e629588..dae1ef6 100644
--- a/media/libstagefright/foundation/ANetworkSession.cpp
+++ b/media/libstagefright/foundation/ANetworkSession.cpp
@@ -662,7 +662,7 @@ status_t ANetworkSession::Session::writeMore() {
         mSawSendFailure = true;
     }
 
-#if 0
+#if 1
     int numBytesQueued;
     int res = ioctl(mSocket, SIOCOUTQ, &numBytesQueued);
     if (res == 0 && numBytesQueued > 50 * 1024) {
diff --git a/media/libstagefright/wifi-display/MediaSender.cpp b/media/libstagefright/wifi-display/MediaSender.cpp
index fef0a65..1bd2724 100644
--- a/media/libstagefright/wifi-display/MediaSender.cpp
+++ b/media/libstagefright/wifi-display/MediaSender.cpp
@@ -242,7 +242,6 @@ status_t MediaSender::queueAccessUnit(
             }
 
             if (minTrackIndex < 0) {
-                adjustAudioQueue();
                 return OK;
             }
 
@@ -284,47 +283,6 @@ status_t MediaSender::queueAccessUnit(
                 ? RTPSender::PACKETIZATION_AAC : RTPSender::PACKETIZATION_H264);
 }
 
-void MediaSender::adjustAudioQueue()
-{
-    for (size_t i = 0; i < mTrackInfos.size(); ++i) {
-        TrackInfo &info = (TrackInfo &)mTrackInfos.itemAt(i);
-        if (info.mAccessUnits.size() > 10) {
-            List< sp<ABuffer> >::iterator lasti = info.mAccessUnits.end();
-            List< sp<ABuffer> >::iterator firsti = info.mAccessUnits.begin();
-            lasti --;
-            sp<ABuffer> first = *firsti;
-            sp<ABuffer> last = *lasti;
-            int64_t lastTimeUs = 0, firstTimeUs = 0;
-            first->meta()->findInt64("timeUs", &firstTimeUs);
-            last->meta()->findInt64("timeUs", &lastTimeUs);
-
-            // if audio timestamp span in the queue large than 300ms.
-            // start to drop audio frames.
-            if ((lastTimeUs - firstTimeUs) > 300000) {
-                ALOGI("adjustAudioQueue timeus:%lld", lastTimeUs - firstTimeUs);
-                int64_t startTimeUs = 0;
-                sp<ABuffer> start;
-                List< sp<ABuffer> >::iterator starti;
-                while (firsti != lasti) {
-                    starti = firsti;
-                    start = *starti;
-                    firsti ++;
-                    int64_t startTimeUs = 0;
-                    start->meta()->findInt64("timeUs", &startTimeUs);
-                    // drop audio timestamp span more than 200ms.
-                    if (lastTimeUs - startTimeUs >= 200000) {
-                        ALOGI("adjustAudioQueue drop timeus:%lld", lastTimeUs - startTimeUs);
-                        info.mAccessUnits.erase(starti);
-                    }
-                    else {
-                        break;
-                    }
-                }//end while
-            }
-        }//end if
-    }//end for
-}
-
 void MediaSender::onMessageReceived(const sp<AMessage> &msg) {
     switch (msg->what()) {
         case kWhatSenderNotify:
diff --git a/media/libstagefright/wifi-display/MediaSender.h b/media/libstagefright/wifi-display/MediaSender.h
index 666f515..bdd1d13 100644
--- a/media/libstagefright/wifi-display/MediaSender.h
+++ b/media/libstagefright/wifi-display/MediaSender.h
@@ -118,7 +118,6 @@ private:
     void notifyInitDone(status_t err);
     void notifyError(status_t err);
     void notifyNetworkStall(size_t numBytesQueued);
-    void adjustAudioQueue();
 
     status_t packetizeAccessUnit(
             size_t trackIndex,
-- 
1.8.0

